plugins/git/git_commit_behind

```diff
@@ -0,0 +1,260 @@
+              conf['git_path'], file=sys.stderr)
+        sys.exit(1)
+
+    for repo_code in repos_conf.keys():
+        logging.debug(' - %s' % repo_code)
+        try:
+            remote_branch = execute_git_command(
+                repos_conf[repo_code],
+                ['rev-parse', '--abbrev-ref', '--symbolic-full-name', '@{u}'])
+            logging.debug('remote_branch = %s' % remote_branch)
+
+            commits_behind = execute_git_command(
+                repos_conf[repo_code],
+                ['rev-list', 'HEAD..%s' % remote_branch, '--count'])
+
+            print('%s.value %d' % (repo_code, int(commits_behind)))
+        except CalledProcessError as e:
+            logging.error('Error executing git command : %s' % str(e))
+        except FileNotFoundError as e:
+            logging.error('Repo not found at path %s' %
+                          repos_conf[repo_code]['path'])
+
+
+def check_update_repos():
+    if len(sys.argv) > 2:
+        max_interval = int(sys.argv[2])
+    else:
+        max_interval = 7200
+
+    if len(sys.argv) > 3:
+        probability = int(sys.argv[3])
+    else:
+        probability = 12
+
+    if not os.path.isfile(conf['state_file']):
+        logging.debug('No state file -> updating')
+        do_update_repos()
+    elif os.path.getmtime(conf['state_file']) + max_interval < time.time():
+        logging.debug('State file last modified too long ago -> updating')
+        do_update_repos()
+    elif randint(1, probability) == 1:
+        logging.debug('Recent state, but random matched -> updating')
+        do_update_repos()
+    else:
+        logging.debug('Recent state and random missed -> skipping')
+
+
+def do_update_repos():
+    for repo_code in repos_conf.keys():
+        try:
+            logging.info('Fetching repo %s' % repo_code)
+            execute_git_command(repos_conf[repo_code], ['fetch'])
+        except CalledProcessError as e:
+            logging.error('Error executing git command : %s' % str(e))
+        except FileNotFoundError as e:
+            logging.error('Repo not found at path %s' %
+                          repos_conf[repo_code]['path'])
+    logging.debug('Updating the state file')
+    open(conf['state_file'], 'w')
+
+
+if len(sys.argv) > 1:
+    action = sys.argv[1]
+    if action == 'config':
+        print_config()
+    elif action == 'autoconf':
+        if os.access(conf['git_path'], os.X_OK):
+            test_git = call([conf['git_path'], '--version'], stdout=DEVNULL)
+            if test_git == 0:
+                print('yes')
+            else:
+                print('no (git seems to be broken ?!)')
+        else:
+            print('no (git is missing or not executable)')
+    elif action == 'version':
+        print('Git commit behind Munin plugin, version {0}'.format(
+            plugin_version))
+    elif action == 'update':
+        check_update_repos()
+    elif action:
+        logging.warn('Unknown argument \'%s\'' % action)
```

Comment:
You can improve readability by minimizing escaping with alternating quote characters.